src/daft-core/src/array/ops/date.rs

```diff
@@ -195,4 +219,93 @@ impl TimestampArray {
             Box::new(date_arrow),
         )
     }
+
+    pub fn truncate(&self, interval: &str, start_time: &Option<i64>) -> DaftResult<Self> {
+        let physical = self.physical.as_arrow();
+        let DataType::Timestamp(timeunit, tz) = self.data_type() else {
+            unreachable!("Timestamp array must have Timestamp datatype")
+        };
+        let tu = timeunit.to_arrow();
```

Comment:
We don't need to convert to arrow here, since we just use it for matching I think

Reply:
I think we still need it when we convert raw timestamps to chrono datetimes: https://github.com/Eventual-Inc/Daft/pull/2158/files#diff-2f91d88e82a1f56f4bdb5d96263cc577767653a5c5019573b5a212f3fde34abeR271